### PR TITLE
Genome browser redux hook update

### PR DESCRIPTION
**Pull request by msmanoj**
## Related JIRA Issue(s)
https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-1334

## Description
- Updated the genome-browser related files to use the redux hook

### Other effects
- [ ] I have checked any requirements for Google Analytics
- [x] I have created any required tests
- [ ] The PR has as its final commit a WASM/JS update commit if any Rust files were updated.

## Possible complications
- Using a common mock state for all the tests!

### CODE REVIEW

File: src/ensembl/src/content/app/browser/Browser.test.tsx [hunk 1]
```diff
@@ -59,85 +59,95 @@ jest.mock('ensembl-genome-browser', () => {
...
+let store: ReturnType<typeof mockStore>;
+
+const wrapInRedux = (state: typeof mockState = mockState) => {
+  store = mockStore(state);
+  return render(
+    <Provider store={store}>
+      <Browser />
+    </Provider>
+  );
 };
```
Comment (azangru) [inline src/ensembl/src/content/app/browser/Browser.test.tsx:75]: Names. Here and in other test files. The api of the helper function above is:

`wrapInRedux(state) -> renderResult`.

The immediate question is, what's being wrapped in redux? This name would make sense if the function were receiving a React component that it was wrapping in redux (`wrapInRedux(component) -> renderResult`), but, as proposed, the function, if you read its name, wraps in redux... the state?

The simplest fix would be to rename this function to something like `renderBrowserComponent`. A more difficult fix would be to create a true, generalised, reusable `wrapInRedux` function that would accept a component, its props, and redux state, and return the component wrapped in Redux (might be difficult with typescript).

**Comment by azangru** [comment]
Hypothetically, for the Browser component, we could have done something like this:

```tsx

const ConnectedBrowser = () => {
  // here are all the hooks that are selecting stuff from redux and dispatch actions to redux

  return <Browser {...props}/>
}

// literally the same Browser component as we has when we used mapStateToProps and mapDispatchToProps
const Browser = (props) => {

}

export default ConnectedBrowser; // for use in our app

export Browser; // for testing purposes; no need to wrap it in redux

```

Just a thought. Not insisting on doing this, but it might have made testing easier.

File: src/ensembl/src/content/app/browser/browser-cog/BrowserCogList.test.tsx [hunk 2]
```diff
@@ -15,59 +15,74 @@
...
+    <Provider store={store}>
+      <BrowserCogList />
+    </Provider>
+  );
+};
+
 describe('<BrowserCogList />', () => {
-  afterEach(() => {
-    jest.resetAllMocks();
+  beforeEach(() => {
```
[the remark this answers is not in the document]
Comment (msmanoj) [reply src/ensembl/src/content/app/browser/browser-cog/BrowserCogList.test.tsx:56]: It did. I've reverted these 👍

File: src/ensembl/src/content/app/browser/Browser.test.tsx [hunk 3]
```diff
@@ -59,85 +59,95 @@ jest.mock('ensembl-genome-browser', () => {
...
+        `?focus=${faker.lorem.words()}`,
+        updatedState
+      );
+      const stateWithBrowserNavOpen = set(
+        'browser.browserNav.browserNavOpenState.fake_genome_id_1',
+        true,
+        updatedState
+      );
 
       it('is rendered when props.browserNavOpenState is true', () => {
```
[the remark this answers is not in the document]
Comment (msmanoj) [reply src/ensembl/src/content/app/browser/Browser.test.tsx:130]: Updated the description

File: src/ensembl/tests/fixtures/browser.ts [hunk 4]
```diff
@@ -131,3 +142,121 @@ export const createChrLocationValues = () => {
...
+      location: {
+        pathname: '/species/fake_genome_id_1',
+        query: {},
+        state: {},
+        search: '',
+        hash: '',
+        key: '9vnz4q'
+      },
+      action: 'PUSH'
+    },
```
[the remark this answers is not in the document]
Comment (azangru) [reply src/ensembl/tests/fixtures/browser.ts:219]: Which components do that? Can we change them so that they use the `focus` parameter from `react-router`?

[the remark this answers is not in the document]
Comment (msmanoj) [reply src/ensembl/tests/fixtures/browser.ts]: That's done

File: src/ensembl/src/content/app/browser/Browser.test.tsx [hunk 5]
```diff
@@ -64,12 +64,14 @@ const mockStore = configureMockStore([thunk]);
 
 let store: ReturnType<typeof mockStore>;
 
-const renderComponent = (state: typeof mockState = mockState) => {
+const renderComponent = (state: typeof mockState = mockState, search = '') => {
```
Comment (azangru) [inline src/ensembl/src/content/app/browser/Browser.test.tsx:67]: `const renderComponent = (params: { state: typeof mockState, url: string } = { state: mockState, url: '/' })` ?

That would avoid the awkwardness of having the undefined in the function call (`renderComponent(undefined, '?focus=foo')`).

By the way, why don't we change `?focus=foo` to a more realistic `/genome-browser?focus=foo`?

File: src/ensembl/src/content/app/browser/browser-bar/BrowserBar.test.tsx [hunk 6]
```diff
@@ -15,56 +15,62 @@
...
-import { BrowserBar, BrowserBarProps } from './BrowserBar';
+import { Provider } from 'react-redux';
+import thunk from 'redux-thunk';
+import set from 'lodash/fp/set';
 
-import { ChrLocation } from '../browserState';
-
-import { createEnsObject } from 'tests/fixtures/ens-object';
+import { createMockBrowserState } from 'tests/fixtures/browser';
+import { BrowserBar } from './BrowserBar';
```
Comment (imransl) [inline src/ensembl/src/content/app/browser/browser-bar/BrowserBar.test.tsx:25]: Separate this import?

File: src/ensembl/src/content/app/species-selector/components/species-commit-button/SpeciesCommitButton.test.tsx [hunk 7]
```diff
@@ -19,7 +19,7 @@ import { render } from '@testing-library/react';
 import userEvent from '@testing-library/user-event';
 
 import { SpeciesCommitButton } from './SpeciesCommitButton';
-import { createSelectedSpecies } from 'ensemblRoot/tests/fixtures/selected-species';
+import { createSelectedSpecies } from 'tests/fixtures/selected-species';
```
Comment (imransl) [inline src/ensembl/src/content/app/species-selector/components/species-commit-button/SpeciesCommitButton.test.tsx:22]: Same with the import here

File: src/ensembl/src/content/app/entity-viewer/shared/components/entity-viewer-sidebar/entity-viewer-sidebar-modal/modal-views/EntityViewerDownloads.tsx [hunk 8]
```diff
@@ -28,7 +28,7 @@ import { parseEnsObjectId } from 'src/shared/state/ens-object/ensObjectHelpers';
 import InstantDownloadGene from 'src/shared/components/instant-download/instant-download-gene/InstantDownloadGene';
 
 import styles from './EntityViewerDownloads.scss';
-import { GeneMetadata } from 'ensemblRoot/src/shared/types/thoas/metadata';
+import { GeneMetadata } from 'src/shared/types/thoas/metadata';
```
Comment (imransl) [inline src/ensembl/src/content/app/entity-viewer/shared/components/entity-viewer-sidebar/entity-viewer-sidebar-modal/modal-views/EntityViewerDownloads.tsx:31]: Import order

File: src/ensembl/src/content/app/browser/zmenu/ZmenuContent.test.tsx [hunk 9]
```diff
@@ -29,37 +32,50 @@ import {
 } from './ZmenuContent';
 
 import { Markup } from './zmenu-types';
```
Comment (imransl) [inline src/ensembl/src/content/app/browser/zmenu/ZmenuContent.test.tsx:34]: Group this with type imports?

File: src/ensembl/src/content/app/browser/track-panel/track-panel-list/TrackPanelList.test.tsx [hunk 10]
```diff
@@ -15,49 +15,63 @@
...
+import { Provider } from 'react-redux';
+import thunk from 'redux-thunk';
+import set from 'lodash/fp/set';
 
-import { TrackPanelList, TrackPanelListProps } from './TrackPanelList';
+import { createMockBrowserState } from 'tests/fixtures/browser';
+
+import { TrackPanelList } from './TrackPanelList';
 
 import { createEnsObject } from 'tests/fixtures/ens-object';
```
Comment (imransl) [inline src/ensembl/src/content/app/browser/track-panel/track-panel-list/TrackPanelList.test.tsx:28]: Group this with the import at line 24?

File: src/ensembl/src/content/app/browser/browser-nav/BrowserNavBarMain.tsx [hunk 11]
```diff
@@ -15,9 +15,8 @@
  */
 
 import React, { useState } from 'react';
-import { connect } from 'react-redux';
+import { useSelector } from 'react-redux';
 
-import { RootState } from 'src/store';
 import { getBreakpointWidth } from 'src/global/globalSelectors';
 import { BreakpointWidth } from 'src/global/globalConfig';
```
Comment (imransl) [inline src/ensembl/src/content/app/browser/browser-nav/BrowserNavBarMain.tsx:21]: Not part of this PR but can we separate this import as well?

File: src/ensembl/tests/fixtures/browser.ts [hunk 12]
```diff
@@ -17,12 +17,23 @@
...
+  createGenomeKaryotype
+} from 'tests/fixtures/genomes';
+
+import { RegionValidationResponse } from 'src/content/app/browser/browserHelper';
+import { Markup } from 'src/content/app/browser/zmenu/zmenu-types';
+import { TrackSet } from 'src/content/app/browser/track-panel/trackPanelConfig';
+import { Strand } from 'src/shared/types/thoas/strand';
+import { LoadingState } from 'src/shared/types/loading-state';
+import { createTrackStates } from 'tests/fixtures/track-panel';
+import { BreakpointWidth } from 'ensemblRoot/src/global/globalConfig';
```
Comment (jyothishnt) [inline src/ensembl/tests/fixtures/browser.ts:36]: ensemblRoot

**Comment by imransl** [comment]
@msmanoj Apart from the import order and conflicts, I am happy with this PR.

File: src/ensembl/tests/fixtures/browser.ts (hunk 12 above)
Comment (imransl) [reply src/ensembl/tests/fixtures/browser.ts:36]: Also the import order

File: src/ensembl/tests/fixtures/genomes.ts [hunk 13]
```diff
@@ -22,6 +21,7 @@ import {
   GenomeTrackCategory,
   GenomeKaryotypeItemType
 } from 'src/shared/state/genome/genomeTypes';
+import { EnsObjectTrack } from 'ensemblRoot/src/shared/state/ens-object/ensObjectTypes';
```
[the remark this answers is not in the document]
Comment (msmanoj) [reply src/ensembl/tests/fixtures/genomes.ts]: That's done

File: src/ensembl/src/content/app/browser/browser-nav/BrowserNavBarMain.tsx (hunk 11 above)
Comment (msmanoj) [reply src/ensembl/src/content/app/browser/browser-nav/BrowserNavBarMain.tsx]: That's done

File: src/ensembl/src/content/app/entity-viewer/gene-view/components/proteins-list/proteins-list-item-info/ProteinsListItemInfo.test.tsx [hunk 14]
```diff
@@ -20,7 +20,7 @@ import userEvent from '@testing-library/user-event';
 import times from 'lodash/times';
 
 import { ProteinExternalReferenceGroup } from './ProteinsListItemInfo';
-import { createExternalReference } from 'ensemblRoot/tests/fixtures/entity-viewer/external-reference';
+import { createExternalReference } from 'tests/fixtures/entity-viewer/external-reference';
```
[the remark this answers is not in the document]
Comment (msmanoj) [reply src/ensembl/src/content/app/entity-viewer/gene-view/components/proteins-list/proteins-list-item-info/ProteinsListItemInfo.test.tsx:22]: That's done

File: src/ensembl/src/content/app/entity-viewer/shared/components/entity-viewer-sidebar/entity-viewer-sidebar-modal/modal-views/EntityViewerDownloads.tsx (hunk 8 above)
Comment (msmanoj) [reply src/ensembl/src/content/app/entity-viewer/shared/components/entity-viewer-sidebar/entity-viewer-sidebar-modal/modal-views/EntityViewerDownloads.tsx:30]: That's done

File: src/ensembl/src/content/app/species-selector/components/species-commit-button/SpeciesCommitButton.test.tsx (hunk 7 above)
Comment (msmanoj) [reply src/ensembl/src/content/app/species-selector/components/species-commit-button/SpeciesCommitButton.test.tsx]: That's done
